app/src/test/java/org/p2p/wallet/striga/onboarding/StrigaOnboardingPresenterTest.kt

```diff
@@ -0,0 +1,166 @@
+package org.p2p.wallet.striga.onboarding
+
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.Before
+import org.junit.Test
+import kotlin.test.assertEquals
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
+import org.p2p.wallet.auth.repository.Country
+import org.p2p.wallet.auth.repository.CountryRepository
+import org.p2p.wallet.infrastructure.dispatchers.CoroutineDispatchers
+import org.p2p.wallet.striga.repository.StrigaPresetDataLocalRepository
+import org.p2p.wallet.utils.UnconfinedTestDispatchers
+import org.p2p.wallet.utils.back
+import org.p2p.wallet.utils.front
+import org.p2p.wallet.utils.mutableListQueueOf
+
+private val UnsupportedCountry = Country(
+    name = "Turkey",
+    flagEmoji = "🇹🇷",
+    code = "tr"
+)
+
+private val SupportedCountry = Country(
+    name = "United Kingdom",
+    flagEmoji = "🇬🇧",
+    code = "gb"
+)
+
+@ExperimentalCoroutinesApi
+class StrigaOnboardingPresenterTest {
+
+    @MockK
+    lateinit var countryRepository: CountryRepository
+
+    @MockK
+    lateinit var strigaPresetDataLocalRepository: StrigaPresetDataLocalRepository
+
+    private val dispatchers: CoroutineDispatchers = UnconfinedTestDispatchers()
+
+    @Before
+    fun setUp() {
+        MockKAnnotations.init(this)
```

Comment:
aren't annotation is like old-style java? how does it work?

Reply:
`setUp` and `tearDown` are classic names; `@Before` and `@After` controlled by junit, setUp called once before tests, tearDown - after all tests for cleanup resources

Reply:
I meant MockkAnnotations

Reply:
It uses reflection to inject values into fields, then it does the same as `val myDep = mockk()`